articles/actions.md

```diff
@@ -0,0 +1,437 @@
+
+To free up resources, the server should discard the result after a configurable timeout period.
+The timeout can be set as part of options to the action server.
+If the timeout is configured to have value **-1**, then goal results will be "kept forever" (until the action server shuts down).
+If the timeout is configured to have value **0**, then goal results are discarded immediately (after responding to any pending result requests).
+
+### Goal Status Topic
+
+* **Direction**: server publishes
+* **Content**: list of in-progress goals with goal ID, time accepted, and an enum indicating the status
+
+This topic is published by the server to broadcast the status of goals it has accepted.
+The purpose of the topic is for introspection; it is not used by the action client.
+Messages are published when transitions from one status to another occur.
+
+The QoS settings for an action server can be configured by the user.
+The default QoS settings for a DDS middleware should be TRANSIENT LOCAL with a history size of 1.
+This allows new subscribers to always get the latest state.
+
+The possible statuses are:
+
+- *Accepted* - The goal has been accepted by the action server
+- *Executing* - The goal is currently being executing by the action server
+- *Canceling* - The action server will try to cancel the indicated goal
+- *Succeeded* - The action server successfully reached the goal
+- *Aborted* - The action server failed reached the goal
+- *Canceled* - The action server successfully canceled the goal
+
+### Feedback Topic
+
+- **Direction**: server publishes
+- **Content**: goal ID, user defined feedback message
+
+This topic is published by the server to send progress feedback about the goal that is user-defined as part of the [Action Interface Definition](#action-interface-definition).
+It is up to the author of the action server to decide how often to publish the feedback.
+
+The QoS settings for feedback coming from an action server can be configured by the user.
+It is up to the clients to use compatible QoS settings.
+
+### Client/Server Interaction Examples
+
+Here are a couple of sequence diagrams depicting typical interactions between an action client and action server.
+
+#### Example 1
+
+In this example, the action client request a goal and gets a response from the server accepting the goal (synchronous).
+Upon accepting the goal, the action server starts a user defined execution method for completing the goal.
+Following the goal request, the client makes an asynchronous request for the result.
+The user defined method publishes feedback to the action client as it executes the goal.
+Ultimately, the user defined method populates a result message that is used as part of the result response.
+
+![Action Interaction Example 0](../img/actions/interaction_example_0.png)
+
+#### Example 2
+
+This example is almost identical to the first, but this time the action  client requests for the goal to be canceled mid-execution.
```

Comment:
There should only be one space after `action` in this sentence.

Reply:
Fixed: f4b59fa